src/components/checkbox/checkbox.test.jsx

```diff
@@ -0,0 +1,233 @@
+import userEvent from '@testing-library/user-event';
+import Checkbox from './checkbox.jsx';
+
+describe('Checkbox', () => {
+  const name = 'You good?';
+  const requiredProps = {
+    id: 'fart',
+    label: name,
+  };
+
+  it('has defaults', () => {
+    const ref = createRef();
+    render(<Checkbox {...requiredProps} ref={ref} />);
+    expect(document.body).toMatchSnapshot();
+    expect(ref.current).toMatchSnapshot();
+  });
+
+  describe('className API', () => {
+    it('sets <input> classname', () => {
+      const className = 'kerfuffle';
+      render(<Checkbox {...requiredProps} className={className} />);
+      expect(screen.getByRole('checkbox', { name })).toHaveClass(className);
+    });
+  });
+
+  describe('cssContainer API', () => {
+    it('sets input container className', () => {
+      render(<Checkbox {...requiredProps} cssContainer="test-class" />);
+      expect(screen.getByRole('presentation')).toHaveClass('test-class');
+    });
+  });
+
+  describe('id API', () => {
+    it('sets the id', () => {
+      render(<Checkbox {...requiredProps} id="hello-mario" />);
+      expect(screen.getByRole('checkbox', { name })).toHaveAttribute('id', 'hello-mario');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} ref={ref} id="hello-mario" />);
+      expect(ref.current.id).toBe('hello-mario');
+    });
+  });
+
+  describe('label API', () => {
+    it('can be set', () => {
+      render(<Checkbox {...requiredProps} label="Unique label" />);
+      expect(screen.getByRole('checkbox', { name: 'Unique label' })).toBeInTheDocument();
+    });
+  });
+
+  describe('name API', () => {
+    it('sets the name attribute', () => {
+      render(<Checkbox {...requiredProps} name="test-name" />);
+      expect(screen.getByRole('checkbox', { name })).toHaveAttribute('name', 'test-name');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} name="unique-name" ref={ref} />);
+      expect(ref.current.name).toBe('unique-name');
+    });
+  });
+
+  it('sets the onblur handler', () => {
+    const onBlur = jest.fn();
+    render(<Checkbox {...requiredProps} onBlur={onBlur} />);
+    fireEvent.blur(screen.getByRole('checkbox', { name }));
+    expect(onBlur.mock.calls.length).toEqual(1);
+  });
+
+  describe('onChange API', () => {
+    it('sets the onChange handler', () => {
+      const onChange = jest.fn();
+      render(<Checkbox {...requiredProps} onChange={onChange} />);
+      userEvent.click(screen.getByRole('checkbox', { name }));
+      expect(onChange.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('onFocus API', () => {
+    it('sets the onFocus handler', () => {
+      const onFocus = jest.fn();
+      render(<Checkbox {...requiredProps} onFocus={onFocus} />);
+      fireEvent.focus(screen.getByRole('checkbox', { name }));
```

Comment:
We don't want to have test coverage with `fireEvent`. Instead always prefer `userEvent` -- our tests should try to reflect user behavior whenever possible.

Reply:
Yeah, but I don't really know how to focus on something using userEvent. How would you do that?

Reply:
`userEvent.click(the_thing)` ! Or you can tab into it: `userEvent.tab()`

Reply:
The idea behind `userEvent` is: "What would a user do?" Think of the actions you need to do to get the component into the desired state. 

In this case, you would say "I would click the label" or "I would tab to it"